core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -310,11 +330,52 @@ public void run() {
     }
   }
 
-  private ScheduledFuture<?> startDeadlineTimer(Deadline deadline) {
+  private ScheduledFuture<?> startDeadlineTimer(final Deadline deadline) {
```

Comment:
It looks like we don't use any part of the `ScheduledFuture` interface. Maybe just return `Future<?>` and then use a pre-existing Future implementation?